tests/e2e/anytinkClient.js

```diff
@@ -8,41 +15,204 @@ class AnythinkClient {
+      method: Method.POST,
+      url: "/api/users",
+      data: { user },
+    });
+    return result.data?.user;
+  }
+
+  async loginUser(email, password) {
+    const result = await this.#apiCall({
+      method: Method.POST,
+      url: "/api/users/login",
+      data: { user: { email, password } },
+    });
+    return result.data?.user;
+  }
+
+  async getUser(callingUser) {
+    const result = await this.#apiCall({ url: "/api/user", callingUser });
+    return result.data?.user;
+  }
+
+  async updateUser(userInfo, callingUser) {
+    const result = await this.#apiCall({
+      method: Method.PUT,
+      url: "/api/user",
+      callingUser,
+      data: { user: userInfo },
+    });
+    return result.data?.user;
+  }
+
+  async createItem(item, callingUser) {
+    const result = await this.#apiCall({
+      method: Method.POST,
+      url: "/api/items",
+      callingUser,
+      data: { item },
+    });
+    return result.data?.item;
+  }
+
+  async deleteItem(slug, callingUser) {
+    await this.#apiCall({
+      method: Method.DELETE,
+      url: `/api/items/${slug}`,
+      callingUser,
+    });
+  }
+
+  async updateItem(slug, item, callingUser) {
+    const result = await this.#apiCall({
+      method: Method.PUT,
+      url: `/api/items/${slug}`,
+      callingUser,
+      data: { item },
+    });
+    return result.data?.item;
+  }
+
+  async getItem(slug, callingUser) {
+    const result = await this.#apiCall({
+      url: `/api/items/${slug}`,
+      callingUser,
+    });
+    return result.data?.item;
+  }
+
+  async favoriteItem(slug, callingUser) {
+    const result = await this.#apiCall({
+      method: Method.POST,
+      url: `/api/items/${slug}/favorite`,
+      callingUser,
+    });
+    return result.data?.item;
+  }
+
+  async unfavoriteItem(slug, callingUser) {
+    const result = await this.#apiCall({
+      method: Method.DELETE,
+      url: `/api/items/${slug}/favorite`,
+      callingUser,
+    });
+    return result.data?.item;
+  }
+
+  async commentOnItem(slug, commenBody, callingUser) {
+    const result = await this.#apiCall({
+      method: Method.POST,
+      url: `/api/items/${slug}/comments`,
+      callingUser,
+      data: { comment: { body: commenBody } },
+    });
+    return result.data?.comment;
+  }
+
+  async deleteComment(slug, commentId, callingUser) {
+    await this.#apiCall({
+      method: Method.DELETE,
+      url: `/api/items/${slug}/comments/${commentId}`,
+      callingUser,
+    });
+  }
+
+  async getComments(slug) {
+    const result = await this.#apiCall({ url: `/api/items/${slug}/comments` });
+    return result.data?.comments;
   }
 
-  async createItem(item) {
-    try {
-      const itemRes = await this.client.post(`/api/items`, { item });
-      return itemRes.data?.item;
-    } catch (error) {
-      console.error("Create Item request failed", error?.response?.data);
-      throw error;
+  async getUserItems(seller, limit, offset, favorited, tag, callingUser) {
```

Comment:
would be nice to be make it named params but probably doesn't matter much

Reply:
Did not want just one or two functions to be with names params, and APIs are easier to use without it